src/components/Lottie/index.tsx

```diff
@@ -23,18 +25,37 @@ function Lottie({source, webStyle, ...props}: Props, ref: ForwardedRef<LottieVie
     useNetwork({onReconnect: () => setIsError(false)});
 
     const [animationFile, setAnimationFile] = useState<string | AnimationObject | {uri: string}>();
+    const [isInteractionComplete, setIsInteractionComplete] = useState(false);
+    const isLastRouteInRHP = isLastRouteRHP();
+    const {isSmallScreenWidth} = useResponsiveLayout();
 
     useEffect(() => {
         setAnimationFile(source.file);
     }, [setAnimationFile, source.file]);
 
+    useEffect(() => {
+        if (!isLastRouteInRHP) {
+            return;
+        }
+
+        const interactionTask = InteractionManager.runAfterInteractions(() => {
+            setIsInteractionComplete(true);
+        });
+
+        return () => {
+            interactionTask.cancel();
+        };
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
+    }, []);
+
     const aspectRatioStyle = styles.aspectRatioLottie(source);
 
     // If the image fails to load, app is in background state, animation file isn't ready, or the splash screen isn't hidden yet,
     // we'll just render an empty view as the fallback to prevent
     // 1. memory leak, see issue: https://github.com/Expensify/App/issues/36645
     // 2. heavy rendering, see issue: https://github.com/Expensify/App/issues/34696
-    if (isError || appState.isBackground || !animationFile || splashScreenState !== CONST.BOOT_SPLASH_STATE.HIDDEN) {
+    // 3. lag on react navigation transitions, see issue: https://github.com/Expensify/App/issues/44812
+    if (isError || appState.isBackground || !animationFile || splashScreenState !== CONST.BOOT_SPLASH_STATE.HIDDEN || (!isInteractionComplete && isLastRouteInRHP && isSmallScreenWidth)) {
```

Comment:
@wildan-m I believe we are applying the change to larger screen as well. So, should we get rid of `isSmallScreenWidth` from the condition. Please let me know if I am missing something.

Reply:
@sobitneupane my [isLastRouteInRHP](https://github.com/Expensify/App/pull/48143/files#diff-9b98479cdfac235633627cb08a9a313e61e100f476ab9d2346620570f275a886) function does not cover current screen, it will be false if the prev screen is not in RHP, i.e. call sign in page from sign in button. I thought that isSmallScreenWidth might still needed.

https://github.com/user-attachments/assets/9d02ce2d-ad56-4fab-a726-95c6c32b2c82

Reply:
What are your thoughts on [using isAnonymous](https://github.com/Expensify/App/pull/48143#issuecomment-2333613910)?

Reply:
@sobitneupane I believe just using `navigationRef?.isReady()` is enough. When loading through a direct link, it's always false, but when the sign-in page modal comes from other screens, it's always true. What's your opinion?

Reply:
@wildan-m It should be fine using isAnonymous if the issue is existent only for anonymous user even is small screen devices. I am not sure about `navigationRef?.isReady()`.

Reply:
@sobitneupane Do you know if there are any other anonymously loaded lottie files besides `Hands.lottie`? I will create a `shouldLoadAfterInteraction` for the Lottie component to prevent the code from affecting other Lottie assets. Let me know if you have any objections.

Reply:
@sobitneupane the code has been updated